spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -356,7 +359,7 @@ object Engine {
       val patternLocalStar = "local\\[\\*\\]".r
       master match {
         case patternLocalN(n) => Some(1, n.toInt)
-        case patternLocalStar(_*) => Some(1, getCoreNumberFromEnv)
+        case patternLocalStar(_*) => Some(1, getNumMachineCores)
```

Comment:
I think it is still a good idea to eliminate the env as much as we can; the error above seems to be an bug (in Spark or BigDL) and we should fix that.